etcdutl/etcdutl/snapshot_command.go

```diff
@@ -110,12 +114,19 @@ func SnapshotRestoreCommandFunc(restoreCluster string,
 	restorePeerURLs string,
 	restoreName string,
 	skipHashCheck bool,
+	revisionBump uint64,
+	markCompacted bool,
 	args []string) {
 	if len(args) != 1 {
 		err := fmt.Errorf("snapshot restore requires exactly one argument")
 		cobrautl.ExitWithError(cobrautl.ExitBadArgs, err)
 	}
 
+	if (revisionBump == 0 && markCompacted) || (revisionBump > 0 && !markCompacted) {
```

Comment:
Do we support only bumping revision, and not marking compacted?

Reply:
No, we only support bumping revision and marking compacted or not bumping revision and not marking compacted. As Marek said above, the flags are separate for clarity, but they need to be used together.

Reply:
@serathius thoughts?

Reply:
Don't understand the argument presented by @ahrtr, I think we are getting into non important details of personal preference. I would recommend to keep the flags separate as it's more clear to user and it's easier later to remove a flag then split the flag if we would decide to support bumping revision without marking it as compacted.

Reply:
Don't think any user-facing change is non-important. Marking compact should never be a must. I am evening think why do we need it on second thought. Please make sure you understand my comment above and let me know if you have confusion.

Reply:
Please read and comment on the original proposal https://github.com/kubernetes/kubernetes/issues/118501, not spread the discussion and design options in just before merging the code just because of "second thought". If we need to change the design, then let's discuss changing the design.

Reply:
There is no change on the design per my previous comment https://github.com/etcd-io/etcd/pull/16029#discussion_r1243910953.  The reason for raising the comment is to make sure we have thought about other possibilities.

This PR fall into the case 2.2 mentioned above, which I don't strongly disagree. If we decide to follow this way, then we may consider removing "`--mark-compact`", and do it (mark compaction) automatically when bumping the revision. Obviously it's clearer & easier to users.

Reply:
The design https://github.com/kubernetes/kubernetes/issues/118501 proposed your case 2.2, two points `Etcd restore operation should support bumping revision by an arbitrary number` and `
Etcd restore operation should support marking old revisions as compacted` are separated and are not a single point. Meaning those are independent.

Reply:
Unfortunately we don't have enough active maintainers and/or reviewers. Each user facing change or etcd core change should have enough reviews and feedback beforing rejecting or merging.

For this case, let's go for 2.2, and let's be aware that  `The only purpose is to make the users be explicitly aware of the operation` . Otherwise, when we are challenged by users, at least we know how to respond.

Reply:
Ok, could we clarify what the action for @dusk125 ? Is this comment addressed, or they should change something?